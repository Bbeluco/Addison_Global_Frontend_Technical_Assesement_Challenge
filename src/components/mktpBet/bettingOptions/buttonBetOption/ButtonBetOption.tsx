import React from "react";
import { TouchableOpacity, Text, StyleSheet } from "react-native";
import { SEPARATOR } from "../../../../constants/selectOptionConstants";
import { ButtonBetOptionType } from "../../../../types/buttonBetOptionType";

function ButtonBetOption({ 
        pushSelectedOptionInfo,
        betOption,
        name,
        price,
        eventIndex,
        marketIndex,
        selectionIndex }: ButtonBetOptionType): React.JSX.Element {

    const buttonIdentifier = `button${SEPARATOR}${eventIndex}${SEPARATOR}${marketIndex}${SEPARATOR}${selectionIndex}`
    return (
        <TouchableOpacity
            onPress={() => pushSelectedOptionInfo(buttonIdentifier)}
            style={[styles.viewArea, styles.btnChoseBet, betOption.includes(buttonIdentifier) && styles.optinChoosen]}
        >
            <Text>{name}</Text>
            <Text>{price}</Text>
        </TouchableOpacity>
    )
}

const styles = StyleSheet.create({
    viewArea: {
        borderWidth: 1,
        borderColor: 'black', 
    },
    btnChoseBet: {
        borderRadius: 10,
        alignItems: 'center',
        width: 100,
        padding: 5,
        margin: 5
    },
    optinChoosen: {
        backgroundColor: 'green'
    }
})

export default ButtonBetOption