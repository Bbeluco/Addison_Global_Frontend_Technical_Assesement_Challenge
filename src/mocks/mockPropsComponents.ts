import { SEPARATOR } from "../constants/selectOptionConstants"

export const betOptionMock = [{ 'option': `button${SEPARATOR}0${SEPARATOR}0${SEPARATOR}0`, 'teamName': 'b', 'titleBet': 'c', 'price': 1 }]

export function pushSelectedOptionInfoMock(option: string, teamName: string, titleBet: string, price: number) {}