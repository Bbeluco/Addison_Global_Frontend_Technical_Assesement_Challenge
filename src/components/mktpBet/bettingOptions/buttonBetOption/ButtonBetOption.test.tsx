import ButtonBetOption from './ButtonBetOption';
import renderer from 'react-test-renderer';
import { pushSelectedOptionInfoMock, betOptionMock } from '../../../../mocks/mockPropsComponents';

describe('<ButtonBetOption />', () => {
    it('should render the component correctly', () => {
        const component = renderer.create(
            <ButtonBetOption 
                pushSelectedOptionInfo={pushSelectedOptionInfoMock}
                betOption={betOptionMock}
                titleBet='Bet123'
                name='To win'
                price={12}
                eventIndex={0}
                marketIndex={0}
                selectionIndex={0}
            />
        ).toJSON();
        expect(component).toMatchSnapshot();
    })
})