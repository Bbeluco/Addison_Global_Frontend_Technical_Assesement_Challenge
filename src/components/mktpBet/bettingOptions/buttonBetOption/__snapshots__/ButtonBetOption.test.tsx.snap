// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MatchTeamsName /> should render the component correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "green",
      "borderColor": "black",
      "borderRadius": 10,
      "borderWidth": 1,
      "margin": 5,
      "opacity": 1,
      "padding": 5,
      "width": 100,
    }
  }
>
  <Text>
    To win
  </Text>
  <Text>
    12
  </Text>
</View>
`;
